api/v1beta1/disruption_types.go

```diff
@@ -317,3 +318,32 @@ func ReadUnmarshal(path string) (*Disruption, error) {
 
 	return &parsedSpec, nil
 }
+
+// RemoveDeadTargets removes targets not found in matchingTargets from the targets list
+func (status *DisruptionStatus) RemoveDeadTargets(matchingTargets []string) {
+	for index := 0; index < len(status.Targets); index++ {
+		if !utils.Contains(matchingTargets, status.Targets[index]) {
+			status.Targets[len(status.Targets)-1], status.Targets[index] = status.Targets[index], status.Targets[len(status.Targets)-1]
+			status.Targets = status.Targets[:len(status.Targets)-1]
```

Comment:
Does this work? We're modifying a list while we're in the middle of traversing it, won't we end up skipping over some indices?

Reply:
I ran some tests and I can confirm this function won't work as is, due to the aforementioned issue

Reply:
Rookie mistake -- will edit asap